win32ss/user/ntuser/menu.c

```diff
@@ -3023,7 +3023,14 @@ static BOOL FASTCALL MENU_ShowPopup(PWND pwndOwner, PMENU menu, UINT id, UINT fl
         if ((y - height) < monitor->rcMonitor.top || y >= monitor->rcMonitor.bottom)
             y = monitor->rcMonitor.bottom - height;
         else
-            y -= height;
+        {
+            adjHgt = y + UserGetSystemMetrics(SM_CYMENUSIZE) +
+                     2 * UserGetSystemMetrics(SM_CYDLGFRAME);
+            if (adjHgt >= monitor->rcMonitor.bottom)
+                y -= height;
+            else
+                y = adjHgt - height;
```

Comment:
Have you verified, by displaying the values of x/y (and width/height) here already, that the menu rectangle is now correct?
I'm asking because the x/y coordinates are further modified later, see the `if (pExclude)` block below.